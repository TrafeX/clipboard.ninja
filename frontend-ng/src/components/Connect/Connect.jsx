import {
  Box,
  Button,
  Flex,
  Heading,
  NumberInput,
  NumberInputField,
  SimpleGrid,
  Spinner,
  Text,
  useColorModeValue,
} from "@chakra-ui/react";
import { useState } from "react";

const Connect = ({ ownRoomNumber, connectedToRoom, connectToRoom }) => {
  const [roomNumber, setRoomNumber] = useState("");

  const subtextColor = useColorModeValue("gray.600", "gray.400");
  const inputColor = useColorModeValue("gray.500", "gray.500");

  const handleConnectToRoom = (event) => {
    event.preventDefault();

    if (roomNumber === "" || isNaN(roomNumber)) {
      return;
    }
    connectToRoom(parseInt(roomNumber, 10));
  };

  const handleChange = (event) => {
    setRoomNumber(event.target.value);
  };

  if (connectedToRoom !== null) {
    return null;
  }

  return (
    <>
      <SimpleGrid columns={{ base: 1, xl: 2 }} alignItems={"baseline"}>
        <Box mb={{ base: 3, xl: 0 }}>
          <Heading as="h3" size="sm" mb={2}>
            This Device ID
          </Heading>
          <Text fontSize={"sm"} color={subtextColor}>
            Enter this Device ID on the other device to connect.
          </Text>

          {ownRoomNumber ? (
            <Text fontSize={"2xl"} fontWeight={"extrabold"} color={"brand.800"}>
              {ownRoomNumber}
            </Text>
          ) : (
            <Text fontSize={"md"} fontWeight={"extrabold"} color={"brand.800"}>
              Waiting for connection to server..
              <Spinner as={"span"} ml={1} size={"sm"} color={"brand.800"} />
            </Text>
          )}
        </Box>
        <Box>
          <Heading as="h3" size="sm" mb={2}>
            Connect to another device
          </Heading>
          <Text fontSize={"sm"} color={subtextColor}>
            Enter the Device ID of the other device below.
          </Text>
          <form onSubmit={handleConnectToRoom}>
            <Flex mt={"10px"}>
              <Flex direction={"column"} align="left" maxW="320">
                {/* <FormLabel htmlFor="roomNr">Enter number</FormLabel> */}
                <NumberInput
                  name="roomNr"
                  defaultValue={roomNumber}
                  isDisabled={ownRoomNumber === null}
                >
                  <NumberInputField
                    onChange={handleChange}
                    _placeholder={{ color: "gray.500" }}
                    borderColor={inputColor}
                    placeholder={"Enter Device ID"}
                  />
                </NumberInput>
                {/* <Flex>
                  <PinInput name="roomNr" defaultValue={roomNumber} disabled={ownRoomNumber === null}>
                    <PinInputField />
                    <PinInputField />
                    <PinInputField />
                    <Text>-</Text>
                    <PinInputField />
                    <PinInputField />
                    <PinInputField />
                  </PinInput>
                </Flex> */}
              </Flex>
              <Button
                type="submit"
                label="Connect"
                colorScheme={"brand"}
                alignSelf="flex-end"
                mx={2}
                disabled={roomNumber === null}
              >
                Connect
              </Button>
            </Flex>
          </form>
        </Box>
      </SimpleGrid>
    </>
  );
};

export default Connect;
