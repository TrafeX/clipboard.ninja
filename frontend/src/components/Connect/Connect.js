// @flow
import React, { useEffect, useState } from 'react';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import muiThemeable from 'material-ui/styles/muiThemeable';
import spinner from 'most-awesome-spinner-ever';

const style = {
  button: {
    margin: 12,
  },
};

const Connect = ({ownRoomNumber, connectedToRoom, connectToRoom, muiTheme}) => {

  const [roomNumber, setRoomNumber] = useState('');
  const [spinnerChar, setSpinnerChar] = useState('');
  const [spinnerId, setSpinnerId] = useState(null);

  useEffect(() => {

    const startSpinner = () => {
      setSpinnerId(spinner((char) => setSpinnerChar(char), 200));
    };

    const stopSpinner = () => {
      if (spinnerId !== null) {
        clearInterval(spinnerId);
      }
    };

    if (ownRoomNumber === null) {
      startSpinner();
    } else {
      stopSpinner();
    }

  }, [ownRoomNumber, spinnerId]);

  const handleConnectToRoom = (event: SyntheticEvent<HTMLButtonElement>) => {
    event.preventDefault();

    if (roomNumber === '' || isNaN(roomNumber)) {
      return;
    }
    connectToRoom(parseInt(roomNumber, 10));
  };

  const handleChange = (event: SyntheticInputEvent<HTMLButtonElement>) => setRoomNumber(event.target.value);

  if (connectedToRoom !== null) {
    return null;
  }

  console.log('Connect: room nr:', ownRoomNumber);
  if (ownRoomNumber === null) {
    ownRoomNumber = `Waiting for server connection.. ${spinnerChar}`;
  }

  return (
    <div>
      <h3>
        Device ID:&nbsp;
        <span style={{color: muiTheme.palette.accent1Color}}>
          <strong>
            {ownRoomNumber}
          </strong>
        </span>
      </h3>
      <p>
        Use the Device ID above to connect to the other device.
        Or enter the Device ID of the other device below.
      </p>
      <form onSubmit={handleConnectToRoom}>
        <TextField
          type="number"
          floatingLabelText="Enter the device ID"
          value={roomNumber}
          onChange={handleChange}
          disabled={ownRoomNumber === null}
        />
        <RaisedButton
          type="submit"
          label="Connect"
          primary={true}
          style={style.button}
          disabled={roomNumber === null}
        />
      </form>
    </div>
  );
}

export default muiThemeable()(Connect);
