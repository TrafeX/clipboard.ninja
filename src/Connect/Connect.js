// @flow
import React, { Component } from 'react';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import muiThemeable from 'material-ui/styles/muiThemeable';
import spinner from 'most-awesome-spinner-ever';

const style = {
  button: {
    margin: 12,
  },
};

class Connect extends Component {
  props: {
    ownRoomNumber: number,
    connectedToRoom: string,
    connectToRoom: (number) => void,
    muiTheme: Object,
  };
  state = {
    roomNumber: '',
    spinner: ' ',
  };
  state: {
    roomNumber: number,
    spinner: string,
  };

  componentDidMount = () => {
    spinner((char) => {
      this.setState({
        spinner: char,
      }
    )}, 200);
  };

  connectToRoom = () => {
    if (this.state.roomNumber === '') {
      return null;
    }
    this.props.connectToRoom(this.state.roomNumber);
  };

  handleChange = (event: Object) => {
    this.setState({
      roomNumber: event.target.value,
    })
  };

  render() {
    if (this.props.connectedToRoom !== '') {
      return null;
    }

    // @todo: Add spacing between first 3 and last 3 numbers
    let ownRoomNumber = this.props.ownRoomNumber;
    if (ownRoomNumber === -1) {
      ownRoomNumber = `Waiting for server connection.. ${this.state.spinner}`;
    }
    return (
      <div>
        <h3>
          Device ID:&nbsp;
          <span style={{color: this.props.muiTheme.palette.accent1Color}}>
            <strong>
              {ownRoomNumber}
            </strong>
          </span>
        </h3>
        <p>
          Use the above Device ID to connect to the other device.<br />
          Or enter the Device ID of the other device below.
        </p>
        <TextField
          type="number"
          floatingLabelText="Enter the device ID"
          value={this.state.roomNumber}
          onChange={this.handleChange}
        />
        <RaisedButton
            label="Connect"
            primary={true}
            style={style.button}
            onClick={() => this.connectToRoom()}
            disabled={this.state.roomNumber === ''}
          />
      </div>
    );
  }
}

export default muiThemeable()(Connect);
