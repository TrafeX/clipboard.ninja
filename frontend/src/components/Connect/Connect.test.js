import React from 'react';
import Connect from './Connect';
import { shallow } from 'enzyme';

it('renders without crashing', () => {

  const wrapper = shallow(
    <Connect connectedToRoom={null}/>
  ).find('Connect').shallow();
  expect(wrapper).toMatchSnapshot();
});
