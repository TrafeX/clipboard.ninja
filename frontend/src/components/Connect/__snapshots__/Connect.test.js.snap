// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <h3>
    Device ID:Â 
    <span
      style={
        Object {
          "color": "#ff4081",
        }
      }
    >
      <strong />
    </span>
  </h3>
  <p>
    Use the Device ID above to connect to the other device. Or enter the Device ID of the other device below.
  </p>
  <form
    onSubmit={[Function]}
  >
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Enter the device ID"
      fullWidth={false}
      multiLine={false}
      onChange={[Function]}
      rows={1}
      type="number"
      underlineShow={true}
      value=""
    />
    <RaisedButton
      disabled={false}
      fullWidth={false}
      label="Connect"
      labelPosition="after"
      primary={true}
      secondary={false}
      style={
        Object {
          "margin": 12,
        }
      }
      type="submit"
    />
  </form>
</div>
`;
